test/OpenSpaceToolkit/Astrodynamics/Trajectory/State/CoordinateSubset.test.cpp

```diff
@@ -132,6 +132,39 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_State_CoordinateSubset, InFrame
 
         EXPECT_EQ(expected, actual);
     }
+
+    // Attempt to transform a non-scalar subset without implementing the inFrame method
+    {
+        const Shared<const CoordinateSubset> nonScalarCoordinateSubset = std::make_shared<CoordinateSubset>("NAME", 3);
+        const Shared<const CoordinateBroker> nonScalarCoordinateBroker =
+            std::make_shared<CoordinateBroker>(Array<Shared<const CoordinateSubset>>({nonScalarCoordinateSubset}));
+
+        const Instant instant = Instant::J2000();
+        const Shared<const Frame> fromFrame = Frame::GCRF();
+        const Shared<const Frame> toFrame = Frame::TEME();
+        VectorXd fullCoordinatesVector(3);
+        fullCoordinatesVector << 1.0e7, -1e7, 5e6;
+
+        EXPECT_THROW(
```

Comment:
Nit: don't most of other tests simply assert that an exception is thrown?

Nothing wrong with asserting the message string. I'm not a big fan because it makes the test much longer and it involves reworking the test for a tangential change in an error message.

Reply:
Not necessarily, I think its more common to assert the full error message because that way you're sure its the correct error being thrown.

And if you make a change to the cpp code that throws the error you can just ctrl+f to change that message across the repo, easy.

Check other examples here
https://github.com/open-space-collective/open-space-toolkit-astrodynamics/blob/main/test/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.test.cpp
https://github.com/open-space-collective/open-space-toolkit-astrodynamics/blob/main/test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp#L1211-L1355